Gems/PhysX/Code/Source/EditorHeightfieldColliderComponent.cpp

```diff
@@ -224,6 +226,22 @@ namespace PhysX
             return;
         }
 
+        // If the change is only about heightfield materials mapping, we can simply update material selection in the heightfield shape
+        if (changeMask == HeightfieldChangeMask::SurfaceMapping)
+        {
+            Physics::MaterialSelection updatedMaterialSelection = m_colliderConfig.m_materialSelection;
+            Utils::SetMaterialsFromHeightfieldProvider(GetEntityId(), updatedMaterialSelection);
+
+            // Make sure the number of slots is the same.
+            // Otherwise the heightfield needs to be rebuilt to support updated indices.
+            if (updatedMaterialSelection.GetMaterialIdsAssignedToSlots().size()
+                == m_colliderConfig.m_materialSelection.GetMaterialIdsAssignedToSlots().size())
```

Comment:
This is because changing the number of surface-to-material mappings can affect the entire heightfield, e.g. if there was no mapping for a given surface, but now there is. 
So we can't simply SetMaterials in that case.